CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -104,10 +148,25 @@ local function AddTool(tool)
 	UpdateLayout()
 end
 
+local humanoidChangedEvent = nil
+local function RegisterHumanoid(humanoid)
+	if humanoidChangedEvent then
+		humanoidChangedEvent:disconnect()
+		humanoidChangedEvent = nil
+	end
+	if humanoid then
+		humanoidChangedEvent = humanoid.Changed:connect(function() UpdateHealth(humanoid) end)
```

Comment:
I could watch for Parent changing.

What happens to the healthbar if you remove the humanoid though? The way it currently functions is that it will just switch to whatever new humanoid you add after. Should the healthbar just disappear if there's NO humanoid?